### PR TITLE
Sort WingetPath as INT

### PR DESCRIPTION
**Pull request by KnifMelti**
Needs to be done also for **Winget-Install-GUI** and **Winget-Install** (install + detect) - if I'm correct...

### CODE REVIEW

File: Winget-AutoUpdate/functions/Get-WingetCmd.ps1 [hunk 1]
```diff
@@ -3,7 +3,7 @@
 Function Get-WingetCmd {
 
     #Get WinGet Path (if admin context)
-    $ResolveWingetPath = Resolve-Path "C:\Program Files\WindowsApps\Microsoft.DesktopAppInstaller_*_x64__8wekyb3d8bbwe"
+    $ResolveWingetPath = Resolve-Path "C:\Program Files\WindowsApps\Microsoft.DesktopAppInstaller_*_x64__8wekyb3d8bbwe" | Sort-Object -Property {$_.Path -as [int]}
```
[the remark this answers is not in the document]
Comment (KnifMelti) [reply Winget-AutoUpdate/functions/Get-WingetCmd.ps1:6]: Feels better!

**Comment by KnifMelti** [comment]
You forgot Invoke-PostUpdateActions.ps1 ;)

**Comment by KnifMelti** [comment]
```
    #Reset Winget Sources
    $ResolveWingetPath = Resolve-Path "$env:programfiles\WindowsApps\Microsoft.DesktopAppInstaller_*_x64__8wekyb3d8bbwe\winget.exe"
    if ($ResolveWingetPath) {
        #If multiple version, pick last one
        $WingetPath = $ResolveWingetPath[-1].Path
        & $WingetPath source reset --force
        Write-Log "-> Winget sources reseted." "green"
    }

```

**Comment by Romanitho** [comment]
Yes, I have to take some time to improve all these little things :/

**Comment by Romanitho** [comment]
If I don't tell wrong, there is no way to install both version (stable and pre-release) anymore ?!

**Comment by KnifMelti** [comment]
Not that I know of...

**Comment by KnifMelti** [comment]
Latest:
image

After upgrade to preview:
image

**Comment by Romanitho** [comment]
So same as me. no multiple versions anymore

**Comment by Romanitho** [comment]
I have another method also:
```
$WingetInfo = (Get-Item (Resolve-Path "$env:ProgramFiles\WindowsApps\Microsoft.DesktopAppInstaller_*_8wekyb3d8bbwe\winget.exe").path).VersionInfo | Sort-Object -Property FileVersionRaw
$Winget = $WingetInfo[-1].FileName
```

**Comment by KnifMelti** [comment]
But, that we can't use to check the available version online (that's the `"tag_name": "v1.6.2771"`) only...

**Comment by Romanitho** [comment]
that's true again

**Comment by Romanitho** [comment]
But it was not used to check online version...

**Comment by Romanitho** [comment]
My final one:
```
try {
    #Get WinGet Path (if admin context)
    try {
        #Get Admin Context Winget Location
        $WingetInfo = (Get-Item "$env:ProgramFiles\WindowsApps\Microsoft.DesktopAppInstaller_*_8wekyb3d8bbwe\winget.exe").VersionInfo | Sort-Object -Property FileVersionRaw
        #If multiple versions, pick most recent one
        $Script:Winget = $WingetInfo[-1].FileName
    }
    catch {
        #Get User context Winget Location
        if (Test-Path "$env:LocalAppData\Microsoft\WindowsApps\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\winget.exe") {
            $Script:Winget = "$env:LocalAppData\Microsoft\WindowsApps\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\winget.exe"
        }
    }
    Write-Host "Winget path: $Winget"
}
catch {
    Write-Host "Winget not installed"
}
```

I'm working on the Winget-install script, to bring back lastest WAU changes to Winget-install repo

**Comment by KnifMelti** [comment]
Nice and tidy!

**Pull request merged**